docs/design-documents/platform/Statistics/NetworkStatistics.md

```diff
@@ -0,0 +1,186 @@
+                                    |+ static int get_each(stats_info_t *stats, int count); |
+                                    `-------------------------------------------------------'
+                                                                |Inherits
+                                                                |
+                            -----------------------------------------------------------------------------------
+                           |                            |                           |                         |
+                           |                            |                           |                         |
+,----------------------------------------.      ,------------------------.      ,--------------------.      ,----.
+|                 <Module_name>Stats     |      |   EMACInterfaceStats   |      |   LWIPStackStats   |  ... |    |
+|----------------------------------------|      `------------------------'      `--------------------'      `----'
+|+ <module>_stats_t _stats;              |                 |Uses                          |Uses
+|----------------------------------------|                 |                              |
+|+ inline void log_xx();                 |                 |                              |
+|  ...                                   |                 |                              |
+|+ virtual void read_stats(...);         |                 |                              |
+`----------------------------------------'                 |                              |
+               |Uses                                       |                              |
+               |                                           |                              |
+,---------------------------------.     ,---------------------------------.   ,-----------------------------.
+| Class: <Module>                 |     |Class: EMACInterface             |   |Class: LWIP                  |
+|#Object:<Module_name>Stats stats;|     |#Object:EMACInterfaceStats stats;|   |#Object:LWIPStackStats stats;|
+`---------------------------------'     `---------------------------------'   `-----------------------------'
+```
+
+# Detailed design
+
+## StatsList Class
+An abstract class used to create a linked list entry of every module reporting statistics data. Static API's in StatsList class can be used to read statistics data stored in linked list.
+
+#### Data Structures
+
+1. Statistic types for segregation of different modules and can also be used for filtering statistics data.
+```
+/** Stats type */
+typedef enum {
+    STATS_RTOS_GEN = 0x0,
+
+    STATS_PLATFORM_GEN = 0x20,
+
+    STATS_STORAGE_GEN = 0x40,
+
+    STATS_NW_GEN = 0x60,
+    STATS_NW_INTERFACE_EMAC,
+    STATS_NW_INTERFACE_WIFI,
+    STATS_NW_INTERFACE_MESH,
+    STATS_NW_INTERFACE_CELLULAR,
+
+    STATS_NW_STACK_LWIP,
+    STATS_NW_STACK_WIFI,
```

Comment:
What does this mean?

LWIP is implementation of `NetworkStack`
Some WiFi drivers are also implementations of `NetworkStack` and implementations of `WiFiInterface`

But what is WIFI stack? We don't have such API

Reply:
Stack stats and interface stats are separated, though we do not have full WIFI stack, but all socket information needed for stack stats will be added to separate structure then interface stats